python/modules/qcdSmearProducer.py

```diff
@@ -1,95 +1,344 @@
 #!/usr/bin/env python
 import os, sys
 import ROOT
+from array import array
 ROOT.PyConfig.IgnoreCommandLineOptions = True
 from importlib import import_module
+from os import system, environ
 
 from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection, Object
 from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
 from PhysicsTools.NanoAODTools.postprocessing.tools import deltaPhi, deltaR, closest
+from PhysicsTools.NanoAODTools.postprocessing.framework.treeReaderArrayTools import *
+from rootpy.tree import Tree, TreeModel, IntCol, FloatArrayCol
+
 
 class qcdSmearProducer(Module): 
     def __init__(self):
-	self.writeHistFile=True
-	self.metBranchName="MET"
-	self.xBinWidth = 0.01
-	self.minWindow = 0.01
-	self.maxWindow = 0.5;
-	self.nSmears = 100;
-	self.nSmearJets = 2;
-	self.nBootstraps = 50;
-	self.LINEAR_GRANULATED=True
-	self.winType = self.LINEAR_GRANULATED;
-	self.doFlatSampling = True;
-	self.respInputName = "JetResByFlav";
-	self.respFileName = "file:/eos/uscms/store/user/mkilpatr/13TeV/qcd_smearing/resTailOut_combined_filtered_CHEF_puWeight_weight_WoH_NORMALIZED.root"
-	self.respHistoName = "res_b_comp_14"
-	self.targeth = self.loadHisto(self.respFileName,self.respHistoName)
- 
+        self.writeHistFile=True
```

Comment:
Is this the smearing module?

Reply:
Yes it was in the code so it wanted me to update the file